app/src/routes/register/register.vue

```diff
@@ -0,0 +1,139 @@
+<script setup lang="ts">
+import { DEFAULT_AUTH_DRIVER, DEFAULT_AUTH_PROVIDER } from '@/constants';
+import { useServerStore } from '@/stores/server';
+import { useAppStore } from '@directus/stores';
+import { useHead } from '@unhead/vue';
+import { storeToRefs } from 'pinia';
+import { computed, ref, unref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import SsoLinks from '../login/components/sso-links.vue';
+import RegisterForm from './register-form.vue';
+
+withDefaults(
+	defineProps<{
+		logoutReason?: string | null;
+	}>(),
+	{
+		logoutReason: null,
+	},
+);
+
+const { t, te } = useI18n();
+
+const appStore = useAppStore();
+const serverStore = useServerStore();
+const { auth, providerOptions } = storeToRefs(serverStore);
+
+const wasSuccessful = ref(false);
+
+const driver = ref(unref(auth).disableDefault ? unref(providerOptions)?.[0]?.driver : DEFAULT_AUTH_DRIVER);
+const provider = ref(unref(auth).disableDefault ? unref(providerOptions)?.[0]?.value : DEFAULT_AUTH_PROVIDER);
+
+const providerSelect = computed({
+	get() {
+		return provider.value;
+	},
+	set(value: string) {
+		provider.value = value;
+		driver.value = unref(auth).providers.find((provider) => provider.name === value)?.driver ?? DEFAULT_AUTH_DRIVER;
+	},
+});
+
+const authenticated = computed(() => appStore.authenticated);
+
+useHead({
+	title: t('register'),
+});
+</script>
+
+<template>
+	<public-view>
+		<div class="header">
+			<h1 class="type-title">{{ t('register') }}</h1>
+			<div v-if="!authenticated && providerOptions.length > 1" class="provider-select">
+				<v-select v-model="providerSelect" inline :items="providerOptions" label />
+			</div>
+		</div>
+
+		<div v-if="wasSuccessful" class="after-success">
+			<div v-md="t('registration_successful_note')"></div>
+			<v-button large to="/login">{{ t('sign_in') }}</v-button>
+		</div>
+
+		<register-form v-else :provider="provider" @was-successful="wasSuccessful = $event" />
+
+		<sso-links v-if="!authenticated && wasSuccessful == false" :providers="auth.providers" />
```

Comment:
actually, do we want to show sso links on the register page? :thinking:

Reply:
> as those fall under the action of "login", right

Not necessarily if you have public registration enabled for SSO it doubles as both login and registration 🤔 
I wouldnt be against having the SSO buttons on the registration page if configured as such but feels like someone with more UX insight may have more opinions there.

Reply:
Agreed, removing them now

Reply:
> Not necessarily if you have public registration enabled for SSO it doubles as both login and registration 🤔

Yeah, but you actually login to the provider with an existing account there (at least in most cases)